include/util/hex_float.h

```diff
@@ -213,12 +270,310 @@ class HexFloat {
   static const uint32_t fraction_right_shift =
       (sizeof(uint_type) * 8) - num_fraction_bits;
 
+  // The maximum representable normalized exponent.
+  static const int_type max_exponent =
```

Comment:
So for float 16, this is 31-15 = 16.   That's the exponent for infinity and NaN.
Maybe you need a different name or comment, since the example at https://en.wikipedia.org/wiki/Half-precision_floating-point_format#Exponent_encoding
shows Emin  as -14 and Emax as 15.  Those are for normal and subnormal numbers.